### DISCUSSION TITLE
The default-features option in the dependencies defined in the vcpkg.json of a port does not work.

### DISCUSSION THREAD

**Discussion by recsater** [microsoft/vcpkg#30844] [Ideas]
vcpkg.json
```json
"features": {
    "viewer": {
      "description": "Add cvlib_viewer",
      "dependencies": [
        {
          "name": "imgui",
          "features": [
            "opengl3-binding",
            "glfw-binding"
          ]
        },
        {
          "name": "pangolin",
          "default-features": false,
          "features": [
            "gui"
          ]
        }
      ]
    }
```
    
The expected result from this code is that the imgui's opengl-binding and glfw-binding features will be installed successfully, and only the core and gui features of pangolin will be installed, but in reality, additional default-features are installed for pangolin.

I think default-features should also work as expected.

**Comment by dg0yt** [comment]
This has already been subject to heated discussions: It works as designed, even if it is not what (some) users initially expect.

The default behavior is to install the default features, unless the user explicitly requests otherwise for a specific port (dependency). There is no general "minimal mode". There are only explicit requests to minimize particular ports.

If you don't want the default features of pangolin, you have to make pangolin a direct dependency in your user project manifest  with `"default-features": false`. (In classic mode, you would have to add `pangolin[core]`to the install arguments.)
Other uses of `"default-features": false` are just to tell vcpkg that the user can opt-out of that depended-on port's default feature, with regard to a particular dependency relation.
